File: src/python/IntegrationSDK/foundationallm/integration/models/pii_result_anonymized.py
"""
The PIIResultAnonymized class encapsulates the identification
of a PII entity in textual content along with the anonymized
token and the operator that was used to anonymize the text.
"""
from .pii_result import PIIResult

class PIIResultAnonymized(PIIResult):
    """
    The PIIResultAnonymized class encapsulates identifies
    the type of PII entity and location in
    the textual content with the PII anonymized from the
    original request.
    """
    anonymized_text: str
    operator: str

File: src/dotnet/Gatekeeper/Models/Integration/PIIResultAnonymized.cs
using System.Text.Json.Serialization;

namespace FoundationaLLM.Gatekeeper.Core.Models.Integration
{
    /// <summary>
    /// Encapsulates the anonymized PII (personally identifiable information) results.
    /// </summary>
    public class PIIResultAnonymized : PIIResult
    {
        /// <summary>
        /// The anonymized text to be used instead of the original text.
        /// </summary>
        [JsonPropertyName("anonymized_text")]
        public required string AnonymizedText { get; set; }

        /// <summary>
        /// The operator needed to anonymize the original text.
        /// </summary>
        [JsonPropertyName("operator")]
        public required string Operator { get; set; }
    }
}
